tests/test_discoverer.py

```diff
@@ -16,3 +19,33 @@ def test_on_usr1_signal(discoverer):
     usr2 = discoverer.on_usr1_signal(signal.SIGUSR2)
     assert discoverer.signal_received == False
     assert usr2 == None
+
+
+# Copy a media file to database
+def test_discover_file():
+    discoverer1 = discoverer_run_once()
+    discoverer1.start()
+    subprocess.run(
+        [
+            "cp",
+            "tests/samples/big-buck-bunny-480p-30sec.mp4",
+            "tests/samples/sample-file.mp4",
+        ]
+    )
+    discoverer2 = discoverer_run_once()
+    discoverer2.start()
+    result = subprocess.run(
+        ["sqlite3", "flumes-test.db", "select name from files where id=2;"],
+        capture_output=True,
+        text=True,
+        cwd="tests",
+    )
+    assert result.stdout[:-1] == "sample-file.mp4"
+    assert result.stderr == ""
+    subprocess.run(["rm", "tests/samples/sample-file.mp4"])
+    subprocess.run(
+        ["sqlite3", "flumes-test.db", "delete from files where id=2;"],
+        capture_output=True,
+        text=True,
+        cwd="tests",
+    )
```

Comment:
```suggestion
pytest.mark.parametrize("origin_file, destination_file", "database", "file_db_id" \
    ["tests/samples/big-buck-bunny-480p-30sec.mp4", "tests/samples/sample-file.mp4", \
    "flumes-test.db", "2"])
def test_discover_newly_copied_file(origin_file, destination_file, database, expected_file_db_id):
    # Setup
    discoverer_setup = discoverer_run_once()
    discoverer_setup.start()

    # Test
    shutil.copy2(
        origin_file,
        destination_file
        )

    discoverer_test = discoverer_run_once()
    discoverer_test.start()

    db_select_result = subprocess.run(
        ["sqlite3", database, "select name from files where id={};".format(expected_file_db_id)],
        capture_output=True,
        text=True,
        cwd="tests",
    )

    db_select_result.check_returncode()
    assert db_select_result.stdout[:-1] == os.path.basename(destination_file)
    assert db_select_result.stderr == ""

    # Teardown
    os.remove(destination_file)

    db_delete_result = subprocess.run(
        ["sqlite3", database, "delete from files where id={};".format(expected_file_db_id)],
        capture_output=True,
        text=True,
        cwd="tests",
    )

    db_delete_result.check_returncode()
```
Although the original code is functional i took the initiative of refactoring for the following reasons
- reduce hardcoded parameters and values and replace them with test parameters -> easily extensible tests, clearer input data and expected results
- reduce os dependant code (subprocess.run with shutil functions that make use of the native os functions at runtime)
- added return code checks for subprocess.run() calls to make sure that the operations are successful

Some suggestions
- The database name appears as a magic number here but in reality it is related to the one present in the conftest file. If the database will be the same for all tests let's define it as a variable in conftest so that we can access it from any testfile directly.
If you think more than 1 database will be needed let's modify `discoverer_run_once()` -> `discoverer_run_once(test_database)` to be able to initialise the discoverer with different databases
- Here we are assuming a very concrete expected id based on the assumption that before every execution the database will have only 1 element inside. This is not always guaranteed though (e.g. file deletion might fail and thus next time we launch it we will have 2 files in the path). My strong suggestion here is to access the database at setup section, get and store the id of the last element in it. Like this we reduce the input arguments and guarantee that no matter the number of files in the path we will always be asserting against the correct id (number of elements in database at setup + 1)

Reply:
I really appreciate your comment about refactoring this test. Quite interesting. Thanks!